src/library/export_decl.cpp

```diff
@@ -64,6 +64,19 @@ static environment update(environment const & env, export_decl_env_ext const & e
     return env.update(g_ext->m_ext_id, std::make_shared<export_decl_env_ext>(ext));
 }
 
+environment activate_export_decl(environment const & env, name const & in_ns, export_decl const & e) {
```

Comment:
This change doesn't seem necessary?

Reply:
It is a bug fix. Before, https://github.com/leanprover-community/lean/blob/7b63acc2637ffd3c7ca2b75a893facdc20259336/src/library/export_decl.cpp#L90-L92 called `add_export_decl` which added the modifications to the environment again during `perform`. So essentially, every export decl ended up in every `.olean` file. I factored out the part of the code that is shared with `add_export_decl`, and now `add_export_decl` calls `activate_export_decl`: https://github.com/leanprover-community/lean/blob/7b63acc2637ffd3c7ca2b75a893facdc20259336/src/library/export_decl.cpp#L142-L145

Reply:
:+1: